src/pymor/vectorarrays/interface.py

```diff
@@ -970,6 +970,16 @@ def from_numpy(self, data, ensure_copy=False):
         """
         raise NotImplementedError
 
+    @property
+    def dim(self):
+        return self._dim
+
+    @dim.setter
+    def dim(self, dim):
+        if not isinstance(dim, int):
+            raise TypeError(f'unsupported dtype={type(dim)}')
+        self._dim = dim
```

Comment:
I'm not quite sure, but doesn't this make the dimension of a `VectorSpace` mutable? This should not be allowed ..